File: 프로그래머스/스택,큐_기능개발/기능개발.cpp
#include <string>
#include <iostream>
#include <vector>
#include <queue>
#include <math.h>

using namespace std;

vector<int> solution(vector<int> progresses, vector<int> speeds) {
	vector<int> answer;
	queue <int> tmp;
	int cnt = 1, now = 0;
	for (int i = 0; i < progresses.size(); i++) {

		tmp.push(ceil((100 - progresses[i]) / double(speeds[i])));
	}
	while (!tmp.empty()) {
		if (tmp.size() == 1) {
			answer.push_back(1);
			break;
		}
		int first = tmp.front();
		tmp.pop();
		while (!tmp.empty() && first >= tmp.front()) {
			cnt++;
			tmp.pop();
		}
		answer.push_back(cnt);
		cnt = 1;
	}
	return answer;
}

File: 프로그래머스/스택,큐_기능개발/기능개발.py
import math
def solution(progresses, speeds):
    answer = []
    arr = list([math.ceil((100-progresses[i]) / speeds[i]) for i in range(len(progresses))])
    #print(arr)
    cnt =1
    while len(arr)!=0:
        if len(arr)==1:
            answer.append(1)
            break
            
        first = arr.pop(0)
        while(len(arr)!=0 and first>=arr[0]):
            cnt+=1
            arr.pop(0)
        answer.append(cnt)
        cnt = 1
            
        
    return answer